src/instructlab/model/dk_bench_utils.py

```diff
@@ -0,0 +1,310 @@
+        model_name=model_name, temperature=temperature, system_prompt=model_prompt
+    )
+
+    return server, model_config, openai_client
+
+
+def make_run_dir(output_dir: str) -> str:
+    now = datetime.now()
+    timestamp = now.strftime("%m-%d-%Y_%H-%M-%S-%f")
+    if output_dir.endswith("/"):
+        output_dir = output_dir[:-1]
+    run_dir = f"{output_dir}/job-{timestamp}"
+
+    path_does_not_exist = not os.path.exists(output_dir)
+    path_is_not_dir = not os.path.isdir(output_dir)
+    if path_does_not_exist or path_is_not_dir:
+        run_dir = f"{DEFAULTS.EVAL_DATA_DIR}/job-{timestamp}"
+
+    os.makedirs(run_dir, exist_ok=True)
+    return run_dir
+
+
+def print_header():
+    print("\n")
+    print("# DK-BENCH REPORT")
+
+
+def print_results(result: EvaluationResult, model_name: str):
+    print(f"\n## MODEL: {model_name}\n")
+    total_score = 0
+    question_num = 0
+    for score in result.scores:
+        question_num += 1
+        print(f"Question #{question_num}:     {score['domain_specific_rubrics']}/5")
+        total_score += score["domain_specific_rubrics"]
+
+    average = total_score / len(result.scores)
+    average = round(average, 2)
+    print("----------------------------")
+    print(f"Average Score:   {average}/5")
+    print(f"Total Score:     {total_score}/{question_num*5}\n")
+
+
+def create_excel_results_file(excel_file: str, result: EvaluationResult):
+    summary_df = pd.DataFrame()
+    scores = [score["domain_specific_rubrics"] for score in result.scores]
+    summary_df["scores"] = scores
+
+    question_indices = [f"Q{i + 1}" for i in range(len(summary_df))]
+    question_indices.append("Average")
+    question_indices.append("Total Score")
+    question_indices.append("Median")
+
+    average = summary_df.mean(axis=0, numeric_only=True)
+    total_score = summary_df.sum(axis=0, numeric_only=True)
+    median = summary_df.median(axis=0, numeric_only=True)
+
+    summary_df.loc[len(summary_df)] = average
+    summary_df.loc[len(summary_df)] = total_score
+    summary_df.loc[len(summary_df)] = median
+
+    summary_df.insert(0, "question index", question_indices)
+
+    response_df = result.dataset.to_pandas()
+    response_df["scores"] = scores
+
+    with pd.ExcelWriter(excel_file) as writer:
+        response_df.to_excel(writer, sheet_name="dataset", index=False)
+        summary_df.to_excel(writer, sheet_name="Summary", index=False)
+    return summary_df
+
+
+def write_results(
+    result: EvaluationResult, file_formats: str, output_dir: str, model_name: str
+):
+    response_df = result.dataset.to_pandas()
+    response_df["model_name"] = model_name
+
+    scores = [score["domain_specific_rubrics"] for score in result.scores]
+    response_df["scores"] = scores
+
+    now = datetime.now()
+    timestamp = now.strftime("%m-%d-%Y_%H-%M-%S-%f")
+    response_df["evaluation_run"] = f"run-{timestamp}"
+
+    print("Responses and scores written to:")
+    for fmt in file_formats:
```

Comment:
If the goal here is to validate that we can write to the target format, we should be performing this check _before_ the user runs these evaluations, before we run anything else.

Reply:
This way they don't waste a bunch of money and rate limits only to be told that their requested outputs are in an unrecognized format

Reply:
Great point!